proxy/pkg/middleware/accesslog.go

```diff
@@ -0,0 +1,31 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/middleware"
+	"github.com/owncloud/ocis/ocis-pkg/log"
+)
+
+// AccessLog is a middleware to log http requests at info level logging.
+func AccessLog(logger log.Logger) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			start := time.Now()
+			wrap := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
+			next.ServeHTTP(wrap, r)
+
+			logger.Info().
```

Comment:
Shouldn't the log be called before `next.ServeHTTP`?

Reply:
no, the log needs to know the duration so it has to happen after `next.ServeHTTP(wrap, r)`